src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/client/WorkforceAllocationsToDeliusApiClient.kt

```diff
@@ -16,93 +20,80 @@ import uk.gov.justice.digital.hmpps.hmppsworkload.domain.CaseType
 
 class WorkforceAllocationsToDeliusApiClient(private val webClient: WebClient) {
 
-  fun choosePractitioners(crn: String, teamCodes: List<String>): Mono<ChoosePractitionerResponse> {
+  suspend fun choosePractitioners(crn: String, teamCodes: List<String>): ChoosePractitionerResponse? {
     return webClient
       .get()
       .uri("/allocation-demand/choose-practitioner?crn=$crn&teamCode=${teamCodes.joinToString(separator = ",")}")
-      .retrieve()
-      .onStatus(
-        { httpStatus -> HttpStatus.NOT_FOUND == httpStatus },
-        { Mono.error(MissingChoosePractitioner("No choose practitioner found for $crn $teamCodes")) }
-      )
-      .bodyToMono(ChoosePractitionerResponse::class.java)
-      .onErrorResume { ex ->
-        when (ex) {
-          is MissingChoosePractitioner -> Mono.empty()
-          else -> Mono.error(ex)
+      .awaitExchangeOrNull { response ->
+        when (response.statusCode()) {
+          HttpStatus.OK -> response.awaitBody()
+          HttpStatus.NOT_FOUND -> null
+          else -> throw response.createExceptionAndAwait()
         }
       }
   }
 
-  fun getPersonByCrn(crn: String): Mono<PersonSummary> {
-    return getPerson(crn, "CRN")
-      .onErrorResume { ex ->
-        when (ex) {
-          is MissingOffenderError -> Mono.just(PersonSummary(crn, Name("Unknown", "", "Unknown"), CaseType.UNKNOWN))
-          else -> Mono.error(ex)
-        }
+  suspend fun getPersonByCrn(crn: String): PersonSummary? {
+    return getPerson(crn, "CRN") { response ->
+      when (response.statusCode()) {
+        HttpStatus.OK -> response.awaitBody()
+        HttpStatus.NOT_FOUND -> PersonSummary(crn, Name("Unknown", "", "Unknown"), CaseType.UNKNOWN)
+        else -> throw response.createExceptionAndAwait()
       }
+    }
   }
 
-  fun getPersonByNoms(noms: String): Mono<PersonSummary> {
-    return getPerson(noms, "NOMS")
-      .onErrorResume { ex ->
-        when (ex) {
-          is MissingOffenderError -> Mono.empty()
-          else -> Mono.error(ex)
-        }
+  suspend fun getPersonByNoms(noms: String): PersonSummary? {
+    return getPerson(noms, "NOMS") { response ->
+      when (response.statusCode()) {
+        HttpStatus.OK -> response.awaitBody()
+        HttpStatus.NOT_FOUND -> null
+        else -> throw response.createExceptionAndAwait()
       }
+    }
   }
 
-  private fun getPerson(identifier: String, identifierType: String): Mono<PersonSummary> {
+  private suspend fun getPerson(identifier: String, identifierType: String, responseHandler: suspend (ClientResponse) -> PersonSummary?): PersonSummary? {
```

Comment:
the only difference here is how it handles NOT_FOUND, I can just have another parameter for `notFoundReturn: PersonSummary?` or something like that

Reply:
hmm.. not at all consistent with the rest of the code but i guess it works 🤷